Mathlib/GroupTheory/SpecificGroups/Cyclic.lean

```diff
@@ -680,3 +680,90 @@ lemma not_isCyclic_iff_exponent_eq_prime [Group α] {p : ℕ} (hp : p.Prime)
   · exact False.elim <| h_cyc <| isCyclic_of_orderOf_eq_card g <| by aesop
 
 end Exponent
+
+section ZMod
+
+open Subgroup AddSubgroup
+
+variable {G H : Type*}
+
+/-- The isomorphism between `(Multiplicative G) / ⟨.ofAdd g⟩` and `Multiplicative (G / ⟨g⟩)`. -/
+def multiplicativeQuotientZpowersZmultiplesMulEquiv [AddCommGroup G] (g : G) :
+    Multiplicative G ⧸ zpowers (Multiplicative.ofAdd g) ≃* Multiplicative (G ⧸ zmultiples g) :=
+  MulEquiv.refl _
+
+lemma multiplicativeQuotientZpowersZmultiplesMulEquiv_mk_eq [AddCommGroup G] (g : G)
+    (h : Multiplicative G) :
+    multiplicativeQuotientZpowersZmultiplesMulEquiv g (QuotientGroup.mk h) =
+    .ofAdd (QuotientAddGroup.mk h) := rfl
+
+lemma multiplicativeQuotientZpowersZmultiplesMulEquiv_symm_mk_eq [AddCommGroup G] (g : G)
+    (h : G) :
+    (multiplicativeQuotientZpowersZmultiplesMulEquiv g).symm (QuotientAddGroup.mk h) =
+    QuotientGroup.mk (.ofAdd h) := rfl
+
+/-- The kernel of `zpowersHom G g` is equal to the subgroup generated by `orderOf g`. -/
+theorem zpowersHom_ker_eq [Group G] (g : G) :
+    (zpowersHom G g).ker = zpowers (Multiplicative.ofAdd ↑(orderOf g)) := by
+  ext n
+  rw [MonoidHom.mem_ker, mem_zpowers_iff, zpowersHom_apply, ← orderOf_dvd_iff_zpow_eq_one]
+  let m : ℤ := n
```

Comment:
This is defeq abuse, and suggests we're missing some lemmas about `Multiplicative`.

`simp_rw [← ofAdd_zsmul]` helps a little here.

Reply:
Just fixed this. Not sure if I'm still abusing the `divides = exists` defeq, but trying to type anything after the `conv_rhs` line naturally results in the error `no goals to be solved`.

Reply:
Never mind, found a better solution.